react-native-reanimated.d.ts

```diff
@@ -147,7 +147,7 @@ declare module 'react-native-reanimated' {
 
     export const SpringUtils: SpringUtils
 
-    export type TransformStyleTypes = TransformsStyle['transform'] extends readonly (infer T)[] ? T : never
+    export type TransformStyleTypes = TransformsStyle['transform'] extends (readonly (infer T)[] | undefined) ? T : never
     export type AdaptTransforms<T> = { [P in keyof T]: Animated.Adaptable<T[P]> }
```

Comment:
so here I would add `T[P] extends string ? string | number : number` so that rotate types work well too.

Reply:
`export type AdaptTransforms<T> = { [P in keyof T]: Animated.Adaptable<T[P] extends string ? string | number : number> }`

Reply:
According to react-native typings 
```
interface RotateTransform {
    rotate: string;
}

interface RotateXTransform {
    rotateX: string;
}

interface RotateYTransform {
    rotateY: string;
}

interface RotateZTransform {
    rotateZ: string;
}
```
Rotates accepts only strings

Reply:
not in reanimated, it accepts both string and number